core/src/main/scala/org/apache/spark/deploy/worker/DriverWrapper.scala

```diff
@@ -66,4 +77,68 @@ object DriverWrapper {
         System.exit(-1)
     }
   }
+
+  // R or Python are not supported in cluster mode so download the jars to the driver side
+  private def setupDependencies(loader: MutableURLClassLoader, userJar: String): Unit = {
+    val packagesExclusions = sys.props.get("spark.jars.excludes").orNull
```

Comment:
Its not exactly the same with the code in SparkSubmit but I can give it a shot. I thought about too.